### PR TITLE
added laplacian_filter file

### PR DESCRIPTION
**Pull request by ojas-wani**
### Describe your change:
Added the Laplacian filter in digital_image_processing > filters.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: digital_image_processing/filters/laplacian_filter.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# @Author  : ojas-wani
+# @File    : laplacian_filter.py
+# @Time    : 10/04/2023
+
+from cv2 import imread, cvtColor, COLOR_BGR2GRAY,  filter2D, imread, imshow, waitKey, CV_64F, BORDER_DEFAULT
+import numpy as np
+from gaussian_filter import gaussian_filter
+
+def my_laplacian(src, ddepth=-1, ksize=3, scale=1, delta=0, borderType='default') -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (ojas-wani) [reply digital_image_processing/filters/laplacian_filter.py]: src: the source image, which should be a grayscale or color image.
ddepth: the desired depth of the destination image, one of np.uint8, np.uint16, np.int16, np.float32 or np.float64.
ksize: the size of the kernel used to compute the Laplacian filter, which can be 1, 3, 5 or 7. A larger kernel size will produce more accurate results.
scale: an optional scaling factor applied to the computed Laplacian values, which can be used to enhance or reduce the effect of the filter.
delta: an optional value added to the computed Laplacian values, which can be used to shift the output image intensity range.
bordertype: an optional flag that specifies how to handle the image borders, which can be one of 'default', 'replicate', 'reflect', or 'constant'.

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/pull/9783/files

File: digital_image_processing/filters/laplacian_filter.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    # Get the shape and depth of the source image
+    height, width = src.shape[:2]
+    src_depth = src.dtype
+
+    # If ddepth is -1, use the same depth as the source image
+    if ddepth == -1:
+        ddepth = src_depth
+
+    # Create a Laplacian kernel matrix according to the ksize
```
Comment (cclauss) [inline digital_image_processing/filters/laplacian_filter.py:44]:
```suggestion
    # Create a Laplacian kernel matrix according to the size
    if ksize not in (1, 3, 5, 7):
        raise ValueError("ksize must be in (1, 3, 5, 7)")
```

File: digital_image_processing/filters/laplacian_filter.py [hunk 3]
```diff
@@ -15,8 +15,8 @@
 from gaussian_filter import gaussian_filter
 import numpy as np
```
Comment (cclauss) [inline digital_image_processing/filters/laplacian_filter.py:17]: Two blank lines before a function definition.  psf/black should do this automatically.
```suggestion


```

File: digital_image_processing/filters/laplacian_filter.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    cvtColor,
+    filter2D,
+    imread,
+    imshow,
+    waitKey,
+)
+from gaussian_filter import gaussian_filter
+import numpy as np
+
+def my_laplacian(ksize: int, src: np.ndarray) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (cclauss) [reply digital_image_processing/filters/laplacian_filter.py]: We require tests.

**Comment by cclauss** [comment]
codespell................................................................Failed
- hook id: codespell
- exit code: 65

digital_image_processing/filters/laplacian_filter.py:77: Deteced ==> Detected

File: digital_image_processing/filters/laplacian_filter.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    BORDER_DEFAULT,
+    COLOR_BGR2GRAY,
+    CV_64F,
+    cvtColor,
+    filter2D,
+    imread,
+    imshow,
+    waitKey,
+)
+from digital_image_processing.filters.gaussian_filter import gaussian_filter
```
Comment (cclauss) [inline digital_image_processing/filters/laplacian_filter.py:16]: I fixed this with `ruff --fix digital_image_processing/filters/laplacian_filter.py` which added one blank line between third-party imports and local imports.

```suggestion

from digital_image_processing.filters.gaussian_filter import gaussian_filter
```

**Comment by ojas-wani** [comment]
My ruff is failing. Can you please help me? I tried both combinations here but still, it is failing.

def my_laplacian(src: np.ndarray, ksize: int) -> np.ndarray:

**Comment by cclauss** [comment]
I just pushed some changes... Let's see how they do.

**Comment by cclauss** [comment]
Yes!!  Just adding a test...

**Comment by ojas-wani** [comment]
thank you for helping

**Comment by ojas-wani** [comment]
So, my pull request can merge. right?

**Comment by cclauss** [comment]
If our test passes, we can merge this thing!

**Comment by cclauss** [comment]
Test is passing!
```
digital_image_processing/filters/gabor_filter.py .                       [ 35%]
digital_image_processing/filters/laplacian_filter.py .                   [ 35%]
digital_image_processing/morphological_operations/dilation_operation.py . [ 36%]
..                                                                       [ 36%]
```

**Review by cclauss — APPROVED** [review]
Nice one!!

**Comment by ojas-wani** [comment]
thank you so much

**Pull request merged**